aggregate/__main__.py

```diff
@@ -149,20 +165,67 @@ def main():
         project=config_values['PROJECT'],
     )
 
+    images = {}
     for function in functions:
-        # Create the function and it's corresponding pubsub and subscription.
-        fxn, _, _ = create_cloud_function(
-            name=function,
-            config_values=config_values,
-            service_account=config_values['GCP_SERVICE_ACCOUNT'],
-            pubsub_topic=pubsub,
-            cloud_service=cloud_service,
-            function_bucket=function_bucket,
-            source_archive_object=source_archive_object,
-            slack_channel=slack_channel,
-        )
+        if config_values.get('TYPE') == 'function':
+            # Create the function and it's corresponding pubsub and subscription.
+            create_cloud_function(
+                name=function,
+                config=config_values,
+                service_account=config_values['GCP_SERVICE_ACCOUNT'],
+                pubsub_topic=pubsub,
+                cloud_service=cloud_service,
+                function_bucket=function_bucket,
+                source_archive_object=source_archive_object,
+                slack_channel=slack_channel,
+            )
+        else:
+            _, image, full_image_name, _, _ = create_cloudrun_job(
+                name=function,
+                config=config_values,
+                service_account=config_values['GCP_SERVICE_ACCOUNT'],
+                pubsub_topic=pubsub,
+                cloudrun_service=cloudrun_service,
+                slack_channel=slack_channel,
+                prebuilt_images=images,
+            )
+            images[config_values['DOCKERIMAGE']] = (image, full_image_name)
+
+
+def configure_docker():
+    sp.check_output(['gcloud', 'auth', 'configure-docker'], stderr=sp.DEVNULL)
+
+
+def get_oauth_token():
+    result = sp.check_output(['gcloud', 'auth', 'print-identity-token'])
```

Comment:
This likely won't work with a service account. Ideally we'd run this as part of CI. Is there an equivalent of [`gcloud run deploy`](https://cloud.google.com/sdk/gcloud/reference/run/deploy) in Pulumi, which builds from source and deploys to Cloud Run? Alternatively, should we build the Docker images outside of Pulumi, similarly to how we handle this for our other repositories?

Reply:
It doesn't look like this function is used in this file - but agreed it would be good to remove this definition.
Does deploying cloud run through Pulumi result in a new version, even if nothing has changed? If so, should we remove old images to avoid cluttering on each release?